[Ticket/14131] Completely overhaul all colors in Colours.css

hanakin opened the pull request:
Completely overhaul all colors in Colours.css to be consistent and standardized. Match all colors to a color styleguide using set color variables.

Tracker: https://tracker.phpbb.com/browse/PHPBB3-14131

PHPBB3-14131

hanakin commented:
!set Styles

Zoddo commented:
See https://github.com/phpbb/phpbb/pull/3866#issuecomment-136151656 :)

hanakin commented:
Ready for review!

Nicofuma commented:
I guess that not every browser supports this feature, right? What about the one we officially support?

Nicofuma commented:
Well actually, I don't really understand what you are doing... just cleaning up the formatting and some weird color definitions?

hanakin commented:
Nothing is supported by a browser those are just the variables I used when I fixed all the incorrect colors in the theme which is the main purpose of this. I removed all the slight variances and replaced all the incorrect colors with the proper ones based on the standardized guide in the comments. I left the comments in for documentation purposes on what colors to use in what situations. Overall reducing the total number of colors in half

Nicofuma commented:
ok, thanks.

By the way, you need to rebase (it conflicts with another of your PRs)

hanakin commented:
yeah working that now there is alot of manual review lol

Nicofuma commented:
@cyberalien Again, could you have a look please? (Yes, there is a lot of styling PRs these days :-))

marc1706 commented on phpBB/styles/prosilver/theme/colours.css:
```diff
@@ -1,24 +1,106 @@
 
-h1 {
-	color: #FFFFFF;
+h1 { 
```
Trailing whitespace should be removed

marc1706 commented:
Your PR seems to contain a lot of trailing whitespace. They're nicely highlighted by github, too.

hanakin commented:
@marc1706 yeah have not really done much work on as every merge affects it drastically it going to have to go as the last merge anyway. So its best to fix it all during the long dynamic rebase

hanakin commented:
!set WIP

cyberalien commented:
Sorry for delay. Color changes look good.

Zoddo commented:
@hanakin : You need to add `PHPBB3-14131` at the end of each commit message for the tracker.

hanakin commented:
ready for review @cyberalien

hanakin commented:
!unset WIP

hanakin commented:
ready for review @prototech should not be any BC issues just helping to standardize colours.

VSEphpbb commented:
Umm,forum background color is awful. Please revert that. There's just a weird wash of blue-i-ness to this that makes phpBB uglier.

hanakin commented:
@VSEphpbb used bg instead of 2 how about now?

VSEphpbb commented:
Actually, I was talking about the body/html background.

hanakin commented:
@VSEphpbb

marc1706 commented:
The post text in the editor is also highlighted in the same color as the outside of the text box border. This looks kind of weird.

hanakin commented:
@marc1706 ill have a look when i get home later tonight

hanakin commented:
@marc1706 try now switched it back to the darker blue

VSEphpbb commented:
Text in post editor should be black. If that's what's being discussed here.

hanakin commented:
hmm can make it default color but its always been dark blue

VSEphpbb commented:
It's always been black. technically, `#333333`
https://github.com/phpbb/phpbb/blob/3.1.x/phpBB/styles/prosilver/theme/colours.css#L973-L987

hanakin commented:
hmm guess the specificity was coded backwords originally not sure why any of the inputs was blue to begin with possibly easier to leave it as all black and remove/deprecate that line.

VSEphpbb commented:
I have to say I'm still not a fan of some of the color changes, or why they are even needed. To my eyes, this PR is just hurting things with the color changes/choices.

That said, still many inconsistencies with font colors, leading to them harder to read. Font colors are too washed out, and sometimes a weird off blue/green when they should be black.

THIS PR:
![screen shot 2015-11-23 at 7 35 48 pm](https://cloud.githubusercontent.com/assets/303711/11357185/c633d7be-9219-11e5-85bf-1ca8f657ec79.png)
![screen shot 2015-11-23 at 7 35 36 pm](https://cloud.githubusercontent.com/assets/303711/11357178/b95a63a0-9219-11e5-9107-87deb4fa3cb3.png)

Original (better):
![screen shot 2015-11-23 at 7 36 41 pm](https://cloud.githubusercontent.com/assets/303711/11357187/cae588de-9219-11e5-8661-b635536e4716.png)
![screen shot 2015-11-23 at 7 36 53 pm](https://cloud.githubusercontent.com/assets/303711/11357188/cdae6450-9219-11e5-98e5-0fd53469159a.png)

hanakin commented:
the problem was the overuse of colors. there were an extreme amount of unique colors being used in the theme making it harder to manage and properly fix things. This is a perfect example of poor design/color usage from the start as you stated should probably be black and I agree but the original was not black. it was a randomly chose tint of blue which had no ties to any of the primary or what should have been the primary colors. A primary color scheme was never chosen, instead the individual colors were chosen on each mockup and thus we ended up with hundreds of unique colors which is a nightmare for web/design. It makes it harder for theme authors to modify or extension authors to add too if they can not easily see what color to use.

There was probably 5-6 different colors for basic text/headers alone. which should not be the case you should only have 1 and possibly a lighter shade of it for subs.

The methods that I used for replace was not optimal for fixing every individual case and but it was the easiest way to standardize the majority.

here is with the secondary dark color to attempt keeping some of the original blue
<img width="733" alt="screenshot 2015-11-24 07 48 30" src="https://cloud.githubusercontent.com/assets/73081/11359813/dda4421a-927f-11e5-841c-2d8f94884d61.png">

VSEphpbb commented:
Still, I don't see any value to messing with Prosilver, when these efforts could be better spent on work for an all new design/theme. This is probably going to make the 3.1 to 3.2 transition for style authors another nightmare (and the 3.0 to 3.2 even more so).

hanakin commented:
it should have little if any impact on authors current themes as no BC is broken all thats changes is the default colors for more consistancy. As for the why yes a new theme is needed and probably better spent working on it but it will be some time until a new theme is finished can be included with the core and completely replaces prosilver to the point where we can not support it anymore given the extent of extensions and styles already based of it.

VSEphpbb commented:
Maybe the bug fixes should be separated from this conceptual change into two PRs (bugs such as message editor text currently being blue)

hanakin commented:
I agree that fixes should be separate and for the most part this should adhere to it. the only deviations is either formating to make it easier to read which can be undone or implemented in 3.2 ie the icon colors.
